sql/src/main/java/io/crate/execution/dml/upsert/TransportShardUpsertAction.java

```diff
@@ -200,6 +201,18 @@ public TransportShardUpsertAction(Settings settings,
                 false,
                 sourceToParse
             );
+            if (indexResult.getResultType() == Engine.Result.Type.MAPPING_UPDATE_REQUIRED) {
+                // Even though the primary waits on all nodes to ack the mapping changes to the master
```

Comment:
great documentation of this behaviour! 👍

Reply:
I only copied it :)